### PR TITLE
Fix issue with no files found if timerange start/end differs in length

### LINKED ISSUE #1878: No files found if timerange start and end have different length
If the start and end time of `timerange` have different length, no files are found. For example
```python
In [1]: from esmvalcore.local import find_files

In [2]: find_files(timerange='1850/2000', short_name='tas', frequency='mon', modeling_realm='atmos', mip='Amon', institute='*', project='CMIP5', dataset='bcc-csm1-1', ensemble='r1i1p1', exp='historical', product='*')
Out[2]: [LocalFile('/home/bandela/climate_data/cmip5/output1/BCC/bcc-csm1-1/historical/mon/atmos/Amon/r1i1p1/v1/tas_Amon_bcc-csm1-1_historical_r1i1p1_185001-201212.nc')]

In [3]: find_files(timerange='185001/2000', short_name='tas', frequency='mon', modeling_realm='atmos', mip='Amon', institute='*', project='CMIP5', dataset='bcc-csm1-1', ensemble='r1i1p1', exp='historical', product='*')
Out[3]: []
```

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

Closes  #1878

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Review by sloosvel — APPROVED** [review]
Looks good to me, it does make more sense to compare between dates that being compared against eachother

File: tests/unit/local/test_select_files.py [hunk 1]
```diff
@@ -20,6 +20,25 @@ def test_select_files():
...
+    ]
+
+    result = _select_files(files, '196201/1967')
+
+    expected = [
+        "pr_Amon_MPI-ESM1-2-HR_historical_r1i1p1f1_gn_196001-196412.nc",
+        "pr_Amon_MPI-ESM1-2-HR_historical_r1i1p1f1_gn_196501-196912.nc",
+    ]
+
+    assert result == expected
```
Comment (valeriupredoi) [inline tests/unit/local/test_select_files.py:39]: can you add a test case here for the reverse situation ie e.g. `result = _select_files(files, '1962/196706')` and for a completely rando combination e.g. `result = _select_files(files, '196203/196711')`?

**Review by valeriupredoi — CHANGES_REQUESTED** [review]
nice, thanks Bouwe and Saskia! One minor suggestion for the test from me, and then I be merging arr :pirate_flag:

**Review by valeriupredoi — APPROVED** [review]
smashing!

**Pull request merged**